app-with-redux-toolkit/src/redux/services.ts

```diff
@@ -1,17 +1,16 @@
 let accountBalance = 0;
 
 export const changeAccountState = async (
-  deltaAmount,
+  deltaAmount: number,
   interestRateCalculation = false
 ) => {
   const randomError = Math.random() <= 0.75;
-  return await new Promise((resolve, reject) => {
+  return await new Promise<number>((resolve, reject) => {
     setTimeout(() => {
```

Comment:
můžeš prosím přepsat ten kód tak,a by nebyl tak moc zanořený?

např:
```
changeAccountState =(...) => {
   await delay(2000)
   return / reject?
}
```

takhle mi to s tím `new Promise` příjde moc komplexní a těžké na čtení